ciao-controller/openstack_image.go

```diff
@@ -114,16 +119,23 @@ func createImageResponse(img imageDatastore.Image) (image.DefaultResponse, error
 }
 
 // ListImages will return a list of all the images in the datastore.
-func (is ImageService) ListImages() ([]image.DefaultResponse, error) {
+func (is ImageService) ListImages(tenant string) ([]image.DefaultResponse, error) {
 	glog.Info("Listing images")
 	response := []image.DefaultResponse{}
 
-	images, err := is.ds.GetAllImages()
+	images, err := is.ds.GetAllImages(tenant)
 	if err != nil {
-		glog.Errorf("Error on listing images: %v", err)
+		glog.Errorf("Error on retrieving images from tenant [%v]: %v", tenant, err)
 		return response, err
 	}
 
+	publicImages, err := is.ds.GetAllImages(string(image.Public))
```

Comment:
yeah, actually, in database, there's a bucket called ``public`` which only contains public images. I'm thinking that I should also create another one for ``internal`` images, in order to make ``internal`` images retrieval faster.